app/data/domain/cdoTaskList.js

```diff
@@ -1,14 +1,52 @@
 const CdoTask = require('./cdoTask')
+const { ActionAlreadyPerformedError } = require('../../errors/domain/actionAlreadyPerformed')
 
 class CdoTaskList {
+  /**
+   * @param {Cdo} cdo
+   */
   constructor (cdo) {
     this._cdo = cdo
   }
 
+  static dateStageComplete (stage) {
+    return stage instanceof Date
+  }
+
+  static dateIsInTheFuture (date) {
+    return date.getTime() > Date.now()
+  }
+
   get _stageOneComplete () {
     return this.applicationPackSent.completed
   }
 
+  get _stageTwoComplete () {
+    return this.applicationPackSent.completed &&
+    this.insuranceDetailsRecorded.completed &&
+    this.microchipNumberRecorded.completed &&
+    this.applicationFeePaid.completed &&
+    this.form2Sent.completed &&
+    this.verificationDateRecorded.completed
+  }
+
+  get cdoSummary () {
+    const microchipNumber = this._cdo.dog.microchipNumber !== null && `${this._cdo.dog.microchipNumber}`.length > 0 ? this._cdo.dog.microchipNumber : undefined
```

Comment:
Can this not be simplified to 
const microchipNumber = this._cdo.dog.microchipNumber ?? undefined